File: src/processors/bestrecipes_processor.py
import json
from storage.database_manager import RecipesDatabaseManager


def process(file_path: str, db_path: str, batch_size: int = 250_000):
    recipes_db_manager = RecipesDatabaseManager(db_path)

    batch: list[tuple[str, str, str]] = []

    with open(file_path, "r", encoding="utf-8") as f:
        for line in f:
            recipe: tuple[str, str, str] = tuple(json.loads(line))

            # Collect batch for DB insertion
            batch.append(recipe)

            if len(batch) >= batch_size:
                print("\nAdding batch")
                recipes_db_manager.bulk_add_recipes(batch)
                print("Batch added")
                batch.clear()

    # Insert any remaining recipes
    if batch:
        recipes_db_manager.bulk_add_recipes(batch)

File: src/processors/bestrecipes_processor.ts
import * as z from "zod";

import { RecipesDatabaseManager } from "~/db_manager";

const RecipeSchema = z.tuple([z.string(), z.string(), z.string()]);

export async function process_jsonl_recipes(
  file_path: string,
  db_path: string,
  batch_size: number = 250_000,
) {
  const recipes_db_manager = new RecipesDatabaseManager(db_path);
  const batch: (readonly [string, string, string])[] = [];
  const file = Bun.file(file_path);
  const lines = (await file.text()).split("\n");
  for (const line of lines) {
    const recipe = RecipeSchema.parse(JSON.parse(line));

    // Collect batch for DB insertion
    batch.push(recipe);
    if (batch.length >= batch_size) {
      console.log("\nAdding batch");
      recipes_db_manager.bulk_add_recipes(batch);
      console.log("Batch added");
      batch.length = 0;
    }
  }

  // Insert any remaining recipes
  if (batch.length) recipes_db_manager.bulk_add_recipes(batch);
}
